### PR TITLE
Change request model and API wiring

### PR DESCRIPTION
**Pull request by currycoder**
This PR adds the ChangeRequest model and wires it up to the create change request API endpoint.

### CODE REVIEW

File: company/migrations/0012_auto_20200306_0959.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.10 on 2020-03-06 09:59
+
+from django.db import migrations, models
```
Comment (lgarvey) [inline company/migrations/0012_auto_20200306_0959.py]: is it worth squashing this?

Comment (currycoder) [reply company/migrations/0012_auto_20200306_0959.py]: Good idea.  I'll do this when I merge the PR.

File: company/models.py [hunk 2]
```diff
@@ -305,3 +312,25 @@ def __str__(self):
...
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4)
+    duns_number = models.CharField(
+        _('Duns number'),
+        max_length=9,
+        unique=True,
+        validators=[duns_number_validator],
+    )
+    changes = JSONField(null=True, blank=True)
```
Comment (alixedi) [inline company/models.py]: Any reason why this is `null=True, blank=True`?

File: company/models.py [hunk 3]
```diff
@@ -305,3 +312,25 @@ def __str__(self):
...
+    status = models.CharField(
+        _('Change Request Status'),
+        choices=ChangeRequestStatus.list(),
+        max_length=100,
+        default=ChangeRequestStatus.pending.name,
+    )
+    created_on = models.DateTimeField(auto_now_add=True)
```
Comment (alixedi) [inline company/models.py]: Would it be useful to also have `submitted_on`?

File: company/serialisers.py [hunk 4]
```diff
@@ -142,12 +134,49 @@ class Meta:
...
     """
     Serialised representation of a ChangeRequest.
     """
-    duns_number = serializers.CharField(max_length=9, min_length=9)
     changes = ChangeRequestChangesSerialiser()
+
+    class Meta:
+        model = ChangeRequest
+        fields = [
+            'id',
```
Comment (alixedi) [inline company/serialisers.py]: Is there some DRF magic that makes this read-only or do we need to include it in `read_only_fields `?

File: company/models.py (hunk 2 above)
Comment (currycoder) [reply company/models.py]: Good point! I think it was a leftover from a copy/paste - Thanks!

File: company/serialisers.py (hunk 4 above)
Comment (currycoder) [reply company/serialisers.py]: There's some DRF magic which makes it `required=False`:

```
>>> from company.serialisers import ChangeRequestSerialiser
>>> c = ChangeRequestSerialiser()
>>> print(repr(c))
ChangeRequestSerialiser():
    id = UUIDField(required=False, validators=[<UniqueValidator(queryset=ChangeRequest.objects.all())>])
    duns_number = CharField(max_length=9, validators=[<django.core.validators.RegexValidator object>, <UniqueValidator(queryset=ChangeRequest.objects.all())>])
    changes = ChangeRequestChangesSerialiser():
        primary_name = CharField(max_length=255, required=False)
        trading_names = ListField(child=CharField(label='Trading names', max_length=255), required=False)
        domain = CharField(allow_blank=True, label='Website domain', max_length=216, required=False)
        address_line_1 = CharField(allow_blank=True, label='Street address', max_length=255, required=False)
        address_line_2 = CharField(allow_blank=True, label='Street address 2', max_length=255, required=False)
        address_town = CharField(allow_blank=True, label='City', max_length=255, required=False)
        address_county = CharField(allow_blank=True, label='State', max_length=255, required=False)
        address_country = CharField(max_length=2, required=False)
        address_postcode = CharField(allow_blank=True, label='Postal code', max_length=20, required=False)
        registered_address_line_1 = CharField(allow_blank=True, label='Street address', max_length=255, required=False)
        registered_address_line_2 = CharField(allow_blank=True, label='Street address 2', max_length=255, required=False)
        registered_address_town = CharField(allow_blank=True, label='City', max_length=255, required=False)
        registered_address_county = CharField(allow_blank=True, label='State', max_length=255, required=False)
        registered_address_country = CharField(max_length=2, required=False)
        registered_address_postcode = CharField(allow_blank=True, label='Postal code', max_length=20, required=False)
        employee_number = IntegerField(allow_null=False, max_value=2147483647, min_value=0, required=False)
        annual_sales = FloatField(allow_null=False, help_text='Annual sales in US dollars', required=False)
        annual_sales_currency = CharField(allow_blank=True, max_length=3, required=False)
    status = ChoiceField(choices=[('pending', 'Pending submission'), ('submitted', 'Submitted')], label='Change Request Status', required=False)
    created_on = DateTimeField(read_only=True)
```

**Comment by currycoder** [comment]
FYI, I'll delete the migrations and re-create a single migration file when I rebase before merge.

**Pull request merged**